service/storage_service.cc

```diff
@@ -1867,6 +1961,146 @@ future<> storage_service::decommission() {
     });
 }
 
+// Runs inside seastar::async context
+void storage_service::run_bootstrap_ops() {
+    auto uuid = utils::make_random_uuid();
+    // TODO: Specify ignore_nodes
+    std::list<gms::inet_address> ignore_nodes;
+    std::list<gms::inet_address> sync_nodes;
+
+    auto start_time = std::chrono::steady_clock::now();
+    for (;;) {
+        sync_nodes.clear();
+        // Step 1: Decide who needs to sync data for bootstrap operation
+        for (const auto& x :_gossiper.endpoint_state_map) {
+            seastar::thread::maybe_yield();
+            const auto& node = x.first;
+            slogger.info("bootstrap[{}]: Check node={}, status={}", uuid, node, _gossiper.get_gossip_status(node));
+            if (node != get_broadcast_address() &&
+                    _gossiper.is_normal_ring_member(node) &&
+                    std::find(ignore_nodes.begin(), ignore_nodes.end(), x.first) == ignore_nodes.end()) {
+                sync_nodes.push_back(node);
+            }
+        }
+        sync_nodes.push_front(get_broadcast_address());
+
+        // Step 2: Wait until no pending node operations
```

Comment:
Shouldn't we check that for all repair-based node operations?
Or is it checked somewhere else for decommission / removenode?

Reply:
I am adding the wait logic here to make start multiple nodes in parallel to survive. We do not need it other operations.